packages/venia-concept/src/ServiceWorker/Utilities/htmlHandler.js

```diff
@@ -1,7 +1,144 @@
 import { HTML_UPDATE_AVAILABLE } from '@magento/venia-ui/lib/constants/swMessageTypes';
+import { parse } from 'node-html-parser';
 
 import { sendMessageToWindow } from './messageHandler';
 
+const generateScriptResource = htmlElement => htmlElement.attributes.src;
+
+const generateLinkResource = htmlElement => htmlElement.attributes.href;
+
+const generateStyleResource = htmlElement => htmlElement.text;
+
+const identity = x => x;
+
+/**
+ * Generates list of resources from the parsed HTML object.
+ *
+ * @param {HTMLElement} parsedHTML
+ *
+ * @returns { scripts: [string], links: [string], styles: [string] }
+ */
+const generateResources = parsedHTML => {
+    const resources = {};
+
+    resources.scripts = parsedHTML
+        .querySelectorAll('script')
+        .map(generateScriptResource)
+        .filter(identity)
```

Comment:
What is this filtering out? What does the `identity` function do?

Reply:
It removes falsy values in this case `null` or `undefined`.

Reply:
Can we give it a better name or add a comment or something? It's not really clear right now.